tools/autograd/derivatives.yaml

```diff
@@ -1676,10 +1676,10 @@
 
 # DO NOT define a backward for to_dense
 # See [Note: Sometimes view derivatives]
-# - name: to_dense(Tensor self, ScalarType? dtype=None) -> Tensor
+# - name: to_dense(Tensor self, ScalarType? dtype=None, bool? masked=False) -> Tensor
 #
-- name: _to_dense(Tensor self, ScalarType? dtype=None) -> Tensor
-  self: to_dense_backward(grad, self)
+- name: _to_dense(Tensor self, ScalarType? dtype=None, bool? masked=False) -> Tensor
```

Comment:
Does it have to be `optional<bool>` and not just `bool`?

Reply:
I think the only reason for keeping the optional is if you want to be able to differentiate when the user passed in a value or nothing (for deprecation purposes for example).
But if you don't, I agree that the optional is not needed.

Reply:
The default value is `False` though, maybe `nullopt` could be better, to exactly differentiate when the user provided the value and probably warn about bc-breaking changes?

Reply:
Using `nullopt` now and the default is now `True` for BC.